From 15f73603c504e30f82783dc3ca4797ff8b24d27c Mon Sep 17 00:00:00 2001
From: Christian Gmeiner <christian.gmeiner@gmail.com>
Date: Sat, 11 Jun 2016 21:21:51 +0200
Subject: [PATCH 2/7] gallium: add PIPE_CAP_MAX_VERTEX_BUFFERS

Add PIPE_CAP_MAX_VERTEX_BUFFERS param, which defaults to 16.

Signed-off-by: Christian Gmeiner <christian.gmeiner@gmail.com>
Signed-off-by: Paul Cercueil <paul@crapouillou.net>
---
 src/gallium/auxiliary/util/u_screen.c | 3 +++
 src/gallium/docs/source/screen.rst    | 1 +
 src/gallium/include/pipe/p_defines.h  | 1 +
 3 files changed, 5 insertions(+)

diff --git a/src/gallium/auxiliary/util/u_screen.c b/src/gallium/auxiliary/util/u_screen.c
index 05dcecd1c27..70eb4a27d74 100644
--- a/src/gallium/auxiliary/util/u_screen.c
+++ b/src/gallium/auxiliary/util/u_screen.c
@@ -401,6 +401,9 @@ u_pipe_screen_get_param_defaults(struct pipe_screen *pscreen,
    case PIPE_CAP_CLIP_PLANES:
       return 1;
 
+   case PIPE_CAP_MAX_VERTEX_BUFFERS:
+      return 16;
+
    default:
       unreachable("bad PIPE_CAP_*");
    }
diff --git a/src/gallium/docs/source/screen.rst b/src/gallium/docs/source/screen.rst
index a081bbc9f80..7e6964282f5 100644
--- a/src/gallium/docs/source/screen.rst
+++ b/src/gallium/docs/source/screen.rst
@@ -564,6 +564,7 @@ The integer capabilities:
   as opposed to writing gl_PointSize for every point.
 * ``PIPE_CAP_TWO_SIDED_COLOR``: Driver supports two-sided coloring.
 * ``PIPE_CAP_CLIP_PLANES``: Driver supports user-defined clip-planes.
+* ``PIPE_CAP_MAX_VERTEX_BUFFERS``: Number of supported vertex buffers.
 
 .. _pipe_capf:
 
diff --git a/src/gallium/include/pipe/p_defines.h b/src/gallium/include/pipe/p_defines.h
index ca98ee55e97..6ee33ca34dc 100644
--- a/src/gallium/include/pipe/p_defines.h
+++ b/src/gallium/include/pipe/p_defines.h
@@ -909,6 +909,7 @@ enum pipe_cap
    PIPE_CAP_POINT_SIZE_FIXED,
    PIPE_CAP_TWO_SIDED_COLOR,
    PIPE_CAP_CLIP_PLANES,
+   PIPE_CAP_MAX_VERTEX_BUFFERS,
 };
 
 /**
-- 
2.24.0

